routes/index.js

```diff
@@ -467,23 +467,104 @@ var routes = {
       });
     }
   },
+  'stripe-charge-failed': function(request, reply) {
+    var event = request.payload;
+
+    if (event.type !== 'charge.failed') {
+      return reply('This hook only processes charge failed events');
+    }
+
+    var charge = event.data.object;
+
+    basket.queue({
+      event_type: event.type,
+      transaction_id: charge.id,
+      failure_code: charge.failure_code
+    });
+
+    return reply("charge failed event processed");
+  },
+  'stripe-charge-refunded': function(request, reply) {
+    var event = request.payload;
+
+    if (event.type !== 'charge.refunded') {
+      return reply('This hook only processes charge.refunded events');
+    }
+
+    var event_type = event.type;
+    var charge = event.data.object;
+    var refund = charge.refunds.data[0];
+
+    var transaction_id = charge.id;
+    var reason = refund.reason;
+    var status = refund.status;
+
+    if (reason === null) {
+      // refunded via dashboard, mark as requested_by_customer
+      reason = 'requested_by_customer';
+    }
+
+    basket.queue({
```

Comment:
This is JS object notation shorthand, the property names will be the name of the variables, and the values will be the values of the variables. 

more details here: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Object_initializer